ui/js/signup.js

```diff
@@ -199,6 +202,13 @@ require(['jquery', 'underscore', 'oae.core'], function($, _, oae) {
             return;
         }
 
+        // Get the email domain for the tenant (if any) so we can validate the email
+        oae.api.admin.getTenant(null, function(err, tenant) {
+            if (!err && tenant.emailDomain) {
+                emailDomain = new RegExp(tenant.emailDomain + '$');
```

Comment:
Also, the `.`'s would be treated as "any character" instead of actual dots since they are unescaped in the regexp.

In the past on the server side, I've checked to see if it is exactly equal, if not, check that it ends with the tenant email domain preceded with a period. If not, then it doesn't match:

e.g., `'caret.cam.ac.uk' === 'cam.ac.uk'`? No. `'caret.cam.ac.uk'.endsWith('.cam.ac.uk')` ? Yes. So it's valid.